mock-postgres-dataprovider-plugin/src/main/java/io/mosip/certify/mockpostgresdataprovider/integration/repository/StatementDataRepositoryImpl.java

```diff
@@ -0,0 +1,24 @@
+package io.mosip.certify.mockpostgresdataprovider.integration.repository;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.persistence.Query;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@ConditionalOnProperty(value = "mosip.certify.postgres.credential.name", havingValue = "StatementData")
```

Comment:
Merge the two interface implementations into one and create a mapping between the `scope` and the query required to fetch the VC.

Reply:
Added the common interface implementation for the repository and the scope-query mapping